iam-persistence/src/main/java/it/infn/mw/iam/persistence/model/IamTotpMfa.java

```diff
@@ -0,0 +1,174 @@
+ * limitations under the License.
+ */
+package it.infn.mw.iam.persistence.model;
+
+import java.io.Serializable;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
+
+@Entity
+@Table(name = "iam_totp_mfa")
+public class IamTotpMfa implements Serializable {
+
+  private static final long serialVersionUID = 1L;
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private Long id;
+
+  @OneToOne()
+  private IamAccount account;
+
+  @Column(name = "secret", nullable = false)
+  private String secret;
+
+  @Column(name = "active", nullable = false)
+  private boolean active;
+
+  @Temporal(TemporalType.TIMESTAMP)
+  @Column(name = "creation_time", nullable = false)
+  private Date creationTime;
+
+  @Temporal(TemporalType.TIMESTAMP)
+  @Column(name = "last_update_time", nullable = false)
+  private Date lastUpdateTime;
+
+  @OneToMany(mappedBy = "totpMfa", cascade = CascadeType.ALL, fetch = FetchType.EAGER,
+      orphanRemoval = true)
+  private Set<IamTotpRecoveryCode> recoveryCodes = new HashSet<>();
+
+  public IamTotpMfa() {}
+
+  public IamTotpMfa(IamAccount account) {
+    this.account = account;
+  }
+
+  public Long getId() {
+    return id;
+  }
+
+  public void setId(Long id) {
+    this.id = id;
+  }
+
+  public IamAccount getAccount() {
+    return account;
+  }
+
+  public void setAccount(final IamAccount account) {
+    this.account = account;
+  }
+
+  public String getSecret() {
+    return secret;
+  }
+
+  public void setSecret(final String secret) {
+    this.secret = secret;
+  }
+
+  public boolean isActive() {
+
+    return active;
+  }
+
+  public void setActive(final boolean active) {
+
+    this.active = active;
+  }
+
+  public Date getCreationTime() {
+
+    return creationTime;
+  }
+
+  public void setCreationTime(final Date creationTime) {
+
+    this.creationTime = creationTime;
+  }
+
+  public Date getLastUpdateTime() {
+
+    return lastUpdateTime;
+  }
+
+  public void setLastUpdateTime(final Date lastUpdateTime) {
+
+    this.lastUpdateTime = lastUpdateTime;
+  }
+
+  public void touch() {
+
+    setLastUpdateTime(new Date());
+  }
+
+  public Set<IamTotpRecoveryCode> getRecoveryCodes() {
+    return recoveryCodes;
+  }
+
+  public void setRecoveryCodes(final Set<IamTotpRecoveryCode> recoveryCodes) {
+    if (this.recoveryCodes.isEmpty()) {
+      this.recoveryCodes = recoveryCodes;
+    } else {
+      this.recoveryCodes.clear();
+      this.recoveryCodes.addAll(recoveryCodes);
+    }
```

Comment:
The code in "else" works also in case recoveryCodes is empty. My suggestion is to replace all the if-else with:
```java
this.recoveryCodes.clear();
this.recoveryCodes.addAll(recoveryCodes);
```

Reply:
Fixed this in the latest commit. Also added some additional setting of creation and update time to the constructors